python/nav/mibs/juniper_alarm_mib.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (C) 2022 Sikt
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.  You should have received a copy of the GNU General Public License
+# along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""JUNIPER-ALARM-MIB MibRetriever"""
+from twisted.internet import defer
+
+from nav.oids import OID
+from nav.smidumps import get_mib
+from nav.mibs.mibretriever import MibRetriever
+
+
+class JuniperAlarmMib(MibRetriever):
+    """JUNIPER-ALARM-MIB MibRetriever"""
+
+    mib = get_mib("JUNIPER-ALARM-MIB")
+
+    @defer.inlineCallbacks
+    def get_yellow_alarm_count(self):
+        """Tries to get a yellow alarm count from a Juniper device"""
+        return self._get_alarm_count("jnxYellowAlarmCount")
```

Comment:
Actually, this shouldn't be decorated with `@defer.inlineCallbacks` (just as `get_red_alarm_count()` isn't).

`_get_alarm_count()` will return a `Deferred` object (since it's decorated with `inlineCallback`), and you'll be fine by just passing that on the caller using return.